nix/pkgs/haskell/haskell.nix

```diff
@@ -57,9 +56,93 @@ let
       "https://github.com/input-output-hk/Win32-network"."94153b676617f8f33abe8d8182c37377d2784bd1" = "0pb7bg0936fldaa5r08nqbxvi2g8pcy4w3c7kdcg7pdgmimr30ss";
       "https://github.com/input-output-hk/hedgehog-extras"."8bcd3c9dc22cc44f9fcfe161f4638a384fc7a187" = "12viwpahjdfvlqpnzdgjp40nw31rvyznnab1hml9afpaxd6ixh70";
     };
+    cabalProjectLocal = lib.optionalString pkgs.stdenv.hostPlatform.isWindows ''
```

Comment:
Is there a reason not to do these things with module option settings?

Reply:
`cabalProjectLocal` is used by `cabal configure`, module option settings are only looked at later.  IIRC these were all things to help make `cabal configure` work when cross compiling for windows.

Reply:
Ugh I see, so we need to do this if we need the options at configure time and can't set them later. I feel like something has gone wrong if we need to do that, but I don't know what else we could do. Worth a comment, anyway.

Reply:
Should the conditional here be a generic cross conditional? i.e. hostPlatform != buildPlatform?